### PR TITLE
fix all test in a branch 'destructuring'

### PR DESCRIPTION
**Pull request by EvgenyStralenya**

### CODE REVIEW

File: destructuring/destructuring-master/test/array.js [hunk 1]
```diff
@@ -4,36 +4,36 @@ var assert = require('assert');
...
     });
     it('chained assignments', () => {
       let c, d;
-      let a, b = [c, d] = [1, 2];
+      let [a, b] = [c, d] = [1, 2];
       assert.deepEqual([a, b, c, d], [1, 2, 1, 2]);
     });
     it('in for-of loop', () => {
-      for (var [a, b] of [[0, 1, 2]]) {}
+      for (var [a, b] of [[0], [1, 2]]) {}
```
Comment (tr3v3r) [inline destructuring/destructuring-master/test/array.js]: or
```
 for (var [,a, b] of [[0, 1, 2]]) {}
```

File: destructuring/destructuring-master/test/string.js [hunk 2]
```diff
@@ -6,15 +6,15 @@ var assert = require('assert');
 
 describe('Destructuring also works on strings', () => {
   it('destructure every character, just as if the string was an array', () => {
-    let a, b, c = 'abc';
+    let [a, b, c] = 'abc'.split('');
```
Comment (tr3v3r) [inline destructuring/destructuring-master/test/string.js]: or
```
 let [a, b, c] =[...'abc']
```

File: destructuring/destructuring-master/test/string.js [hunk 3]
```diff
@@ -6,15 +6,15 @@ var assert = require('assert');
...
     assert.deepEqual([a, b, c], ['a', 'b', 'c']);
   });
   it('missing characters are undefined', () => {
-    const [a, c] = 'ab';
+    const [a, c] = 'ab'.split(',');
```
Comment (tr3v3r) [inline destructuring/destructuring-master/test/string.js]:
```
 const [a, c] = ['ab']
```

Comment (EvgenyStralenya) [reply destructuring/destructuring-master/test/string.js]: adjustment file string.js

File: destructuring/destructuring-master/test/string.js (hunk 2 above)
Comment (EvgenyStralenya) [reply destructuring/destructuring-master/test/string.js]: I agree with this decision

File: destructuring/destructuring-master/test/array.js (hunk 1 above)
Comment (EvgenyStralenya) [reply destructuring/destructuring-master/test/array.js]: I agree with this decision

**Pull request merged**